fpga/Makefile

```diff
@@ -0,0 +1,124 @@
+CONFIG_PACKAGE    := sifive.freedom.everywhere.e300artydevkit
+GENERATOR_PACKAGE := chipyard
+TB                := none # unused
+TOP               := E300ArtyDevKitPlatform
+
+# setup the board to use
+BOARD ?= arty
+
+#########################################################################################
+# misc. directories
+#########################################################################################
+bootrom_dir := $(base_dir)/fpga/bootrom/xip
+fpga_common_script_dir := $(FPGA_DIR)/common/tcl
+fpga_dir := $(base_dir)/fpga/fpga-shells/xilinx
+
+#########################################################################################
+# import other necessary rules and variables
+#########################################################################################
+include $(base_dir)/common.mk
+
+#########################################################################################
+# copy from other directory
+#########################################################################################
+all_vsrcs := \
+	$(sim_vsrcs) \
+	$(base_dir)/generators/sifive-blocks/vsrc/SRLatch.v \
+	$(fpga_dir)/common/vsrc/PowerOnResetFPGAOnly.v \
+	$(build_dir)/$(long_name).rom.v
+
+#########################################################################################
+# build rom for the fpga
+#########################################################################################
+# needed for bootrom makefile
+export BUILD_DIR=$(build_dir)
+export ROCKETCHIP_DIR
+export LONG_NAME=$(long_name)
+export ROMCONF=$(build_dir)/$(long_name).rom.conf
+
+romgen := $(build_dir)/$(long_name).rom.v
+$(romgen): $(sim_vsrcs)
+ifneq ($(bootrom_dir),"")
+	$(MAKE) -C $(bootrom_dir) romgen
+	mv $(build_dir)/rom.v $@
+endif
+
+.PHONY: romgen
+romgen: $(romgen)
+
+#########################################################################################
+# vivado rules
+#########################################################################################
+# combine all sources into single .F
+f := $(build_dir)/$(long_name).vsrcs.F
+$(f): $(sim_common_files) $(all_vsrcs)
+	$(foreach file,$(all_vsrcs),echo "$(file)" >> $@;)
+	cat $(sim_common_files) >> $@
+
+bit := $(build_dir)/obj/$(MODEL).bit
+$(bit): $(romgen) $(f)
+	cd $(build_dir); vivado \
+		-nojournal -mode batch \
+		-source $(fpga_common_script_dir)/vivado.tcl \
+		-tclargs \
+		-top-module "$(MODEL)" \
+		-F "$(f)" \
+		-ip-vivado-tcls "$(shell find '$(build_dir)' -name '*.vivado.tcl')" \
+		-board "$(BOARD)"
+
+.PHONY: bit
+bit: $(bit)
+
+# Build .mcs
+mcs := $(build_dir)/obj/$(MODEL).mcs
+$(mcs): $(bit)
+	cd $(build_dir); vivado -nojournal -mode batch -source $(fpga_common_script_dir)/write_cfgmem.tcl -tclargs $(BOARD) $@ $<
+
+.PHONY: mcs
+mcs: $(mcs)
+
+#########################################################################################
+# mircosemi rules
+#########################################################################################
+# Build Libero project
```

Comment:
Do we know what this is? Does it apply to the subset we are including?

Reply:
I don't know, I just copied this. I think we are just using the Vivado stuff.

Reply:
I think we should delete it for now, rather than keep around dead code (and perhaps make users think we support this).